File: ios/loopr-ios/Models/Token/TokenStatus.swift
//
//  Depth.swift
//  loopr-ios
//
//  Created by kenshin on 2018/3/6.
//  Copyright © 2018年 Loopring. All rights reserved.
//

import Foundation

enum TokenStatus: String, CustomStringConvertible {

    case valid = "VALID"
    case invalid = "INVALID"
    case unknown = "UNKNOWN"

    var description: String {
        switch self {
        case .valid:
            return "valid"
        case .invalid:
            return "invalid"
        case .unknown:
            return "unknown"
        }
    }
}

File: android/walletsdk/src/main/java/leaf/prod/walletsdk/model/token/TokenStatus.java
package leaf.prod.walletsdk.model.token;

/**
 * Created with IntelliJ IDEA.
 * User: kenshin wangchen@loopring.org
 * Time: 2019-03-21 2:37 PM
 * Cooperation: loopring.org 路印协议基金会
 */
public enum TokenStatus {

    VALID("VALID"), INVALID("INVALID"), UNKNOWN("UNKNOWN");

    private final String description;

    TokenStatus(final String text) {
        this.description = text;
    }

    public String getDescription() {
        return description;
    }
}
